### PR TITLE
Deploy to AWS from GHA using OpenTofu

### PR DESCRIPTION
**Pull request by smokestacklightnin**
- [ ] Don't forget to remove debug commits

Find debugging commits with 

```bash
$ git log --oneline | grep -i debug
```

This must be merged after the deploying Docker PR

### CODE REVIEW

File: .github/workflows/deploy-opentofu.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        with:
+          files: |
+            web/deploy/terraform/staging/**
+
+      - name: Check for changes in production deployment
+        id: changed-tofu-production
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/production/**
```
Comment (marcelovilla) [inline .github/workflows/deploy-opentofu.yml]: I see where you're coming from, but we don't really need to check the files for changes. That's exactly what Terraform/OpenTofu will do and it's one of the benefits of having an Infrastructure-as-Code approach. If the configuration in the files matches the infrastructure, nothing will be created, destroyed or updated.

File: .github/workflows/deploy-opentofu.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          tofu -chdir web/deploy/terraform/staging apply -nocolor -auto-approve
+
+      - name: Initialize production resources
+        run: |
+          tofu -chdir web/deploy/terraform/staging init
+
+      - name: Deploy production resources
+        if: steps.changed-tofu-modules.outputs.any_modified == true || steps.changed-tofu-shared.outputs.any_modified == true || steps.changed-tofu-production.outputs.any_modified == true
+        run: |
+          tofu -chdir web/deploy/terraform/production apply -nocolor -auto-approve
```
Comment (marcelovilla) [inline .github/workflows/deploy-opentofu.yml]: We can probably simplify this by using a matrix with the three environments. Something like:
```yaml
strategy:
  matrix:
    environments: ["shared", "staging", "prod"]
```

**Review by marcelovilla — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-opentofu.yml (hunk 1 above)
Comment (smokestacklightnin) [reply .github/workflows/deploy-opentofu.yml]: Addressed

File: .github/workflows/deploy-opentofu.yml (hunk 2 above)
Comment (smokestacklightnin) [reply .github/workflows/deploy-opentofu.yml]: Addressed

**Comment by smokestacklightnin** [comment]
After I finish everything else, I'll look into pretty printing the output of `tofu plan`

**Pull request closed without merging**